test/integration/framework/perf_utils.go

```diff
@@ -33,29 +32,44 @@ const (
 	retries = 5
 )
 
+// NodeTemplate is responsible for creating a v1.Node instance that is ready
+// to be sent to the API server.
+type NodeTemplate interface {
+	// GetNodeTemplate returns a node template for one out of many different nodes.
+	// Nodes with numbers in the range [index, index+count-1] will be created
+	// based on what GetNodeTemplate returns. It gets called multiple times
+	// with a fixed index and increasing count parameters. This number can,
+	// but doesn't have to be, used to modify parts of the node spec like
+	// for example a named reference to some other object.
+	GetNodeTemplate(index, count int) (*v1.Node, error)
+}
+
+// StaticNodeTemplate returns an implementation of NodeTemplate for a fixed node that is the same regardless of the index.
+func StaticNodeTemplate(node *v1.Node) NodeTemplate {
+	return (*staticNodeTemplate)(node)
+}
+
+type staticNodeTemplate v1.Node
```

Comment:
any reason not making it pointer?

```suggestion
type staticNodeTemplate *v1.Node
```